src/spent_proof.rs

```diff
@@ -74,15 +74,21 @@ pub struct SpentProofShare {
     pub spentbook_sig_share: NodeSignature,
 }
 
-impl PartialOrd for SpentProofShare {
-    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
-        Some(self.cmp(other))
+// impl manually to avoid clippy complaint about Hash conflict.
```

Comment:
It'd be better if we can derive both PartialEq, Eq, and Hash. Was one of the fields not hasheable?

Reply:
yes, Hash cannot be derived for SpentProofContent because it includes a Vec<Commitment>, where Commitment is typedef for blstrs::G1Affine, which does not impl Hash.   and of course blstrs is outside our control.

We do have PublicKeyBlstMappable in blst.rs which is a newtype wrapper for G1Affine and impls Hash.  So if we typedef'd Commitment to PublicKeyBlstMappable, then it could be made to work, but is a larger change because any code that uses the Vec<Commitment> has to be modified (builder, spentbook, validation)... I just tried it and its a bit involved to get those building again.

So I think for now let's just go with this.  Hopefully can find some time to clean it up soonish.